src/Hashids.net/Hashids.cs

```diff
@@ -350,16 +368,12 @@ private long[] GetNumbersFrom(string hash)
             if (hashArray.Length == 0)
                 return Array.Empty<long>();
 
-            var i = 0;
-            if (hashArray.Length == 3 || hashArray.Length == 2)
-            {
-                i = 1;
-            }
+            var i = (hashArray.Length is 3 or 2 ) ? 1 : 0;
 
             var hashBreakdown = hashArray[i];
             var lottery = hashBreakdown[0];
 
-            if (lottery == default(char))
+            if (lottery == '\0') /* default(char) == '\0' */
```

Comment:
[Default value expressions are also constant expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#default-value-expressions) for value types in C#, which are evaluated at compile-time already so there's no performance advantage from this.

It's better to stick with `default(char)` for readability.

Reply:
@Jehoel I'll fix this later

Reply:
Yes, I know `default` is a constant-expression, the reason I opted for `'\0'` instead is to signify intent: as `default` is often used in C# with the connotation that the value is unimportant, whereas in this case the zero value is _very significant_ to the algorithm so I wanted to be explicit.